### DISCUSSION TITLE
use @can, @role in client side(second app) app while database is in server side(first app) app

### DISCUSSION THREAD

**Discussion by mirzashahmir02** [spatie/laravel-permission#2247] [Q&A] [answered]
Hi,
First of all, thanks for a great package!

Currently i want to ask a question 

**features of first laravel app for rest api:**----------------------- Server Side App
installed spatie package
database tables User, Role, Permissions, Model has role etc etc
### users table 
john is a super admin and can delete-records
raymond is a admin and can view-record
### Role table
super admin
admin
### Permissions table
view-records
delete records

working of this app is connection with database
and created a login api which is sending a json response of jwt token and auth user who logged in with his roles and permissions

**Features of Second Laravel App for frontend only  (connected to no database):**--------- Client Side App
no spattie package installed in it 
no database connection
logged in by the api of first app and stored the user object in session of this app
now i want to use @role and @can in the blade directives of this app e.g

@role('Super Admin')
<li> delete records </li>
@endrole

this <li> will only be displayed if a super admin(john) logged in 

currently i am using if and else for this. but i want to find a smarter way

**Comment by parallels999** [comment, marked as answer]
Without spattie package there is no `@role`
But you could register your own blade directives for avoid if ... else

**Reply by mirzashahmir02** [reply]
how..?

**Reply by mirzashahmir02** [reply]
by the way i ll search about it thanx buddy

**Comment by erikn69** [comment]
https://github.com/spatie/laravel-permission/blob/bc2227ca5820d337ea5dccde414c17193faee84b/src/PermissionServiceProvider.php#L87-L147

**Reply by mirzashahmir02** [reply]
thanks buddy

> https://github.com/spatie/laravel-permission/blob/bc2227ca5820d337ea5dccde414c17193faee84b/src/PermissionServiceProvider.php#L87-L147
